documentation/optee_design.md

```diff
@@ -208,11 +208,13 @@ is restored it will continue at the next instruction as if this function did a
 normal return. CPU switches to use the temp stack before returning to normal
 world.
 
-## IRQ exit
-IRQ exit occurs when OP-TEE receives an IRQ, which is always handled in normal
-world. IRQ exit is similar to RPC exit but it's `thread_irq_handler()` and
-`elx_irq()` (respectively for ARMv7-A/Aarch32 and for Aarch64) that saves the
-thread state instead. The thread is resumed in the same way though.
+## Foreign interrupt exit
+For ARM GICv2 mode, the foreign interrupts are sent as IRQ.
+Foreign interrupt exit occurs when OP-TEE receives an IRQ, which is always
+handled in normal world. Foreign interrupt exit is similar to RPC exit but
+it's `thread_irq_handler()` and `elx_irq()` (respectively for ARMv7-A/Aarch32
+and for Aarch64) that saves the thread state instead. The thread is resumed
+in the same way though.
```

Comment:
Should add in note about GICv3 ? At least stating OP-TEE does not yet support GICv3 ?
Same remark regarding **interrupt_handling.md**.